dshr/dshr_mod.F90

```diff
@@ -364,91 +417,273 @@ subroutine dshr_mesh_init(gcomp, compid, logunit, compname, model_nxg, model_nyg
   end subroutine dshr_mesh_init
 
   !===============================================================================
-  subroutine dshr_mesh_create_from_grid(filename, compid, mesh, rc)
+  subroutine dshr_mesh_create(filename, scol_mode, scol_lon, scol_lat, &
```

Comment:
It doesn't look like my_task and master_task are used here and if they are only used as in 
my_task == master_task then perhaps that logical should be passed in instead of two integers.

Reply:
done - but you only need to pass in my_task - since master_task is a module variable

Reply:
So it appears that you have decided to do it one way and I have decided to do it the other and we are 
constantly changing it back and forth so that it will never be consistent.  This needs a cleanup.